### ISSUE TITLE
Feature Request: Allow specifying capacity reservations for launch templates

### ISSUE THREAD

**Issue by gazoakley** [terraform-providers/terraform-provider-aws#6323]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS have announced a new Capacity Reservation feature to reserve capacity in AZs without making long term commitments:

* https://aws.amazon.com/about-aws/whats-new/2018/10/Amazon-EC2-now-offers-On-Demand-Capacity-Reservations/
* https://aws.amazon.com/ec2/faqs/#capacityreservations

Reservations can be associated with Launch Templates.

### New or Affected Resource(s)

* `aws_launch_template`

### Potential Terraform Configuration

```hcl
resource "aws_launch_template" "foo" {
  name = "foo"

  capacity_reservation_specification {
    capacity_reservation_preference = "open"
    capacity_reservation_target {
      capacity_reservation_id = "cr-123456789"
    }
  }
}
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateCapacityReservationSpecificationRequest.html

**Comment by bflad** [comment]
Support for this has been merged in and will release with version 1.43.0 of the AWS provider, likely middle of next week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.43.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1430-november-07-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
